csrc/kernel_cache.cpp

```diff
@@ -1149,7 +1150,7 @@ void FusionKernelRuntime::deserialize(
     NVF_ERROR(
         !sg || scheduler_entry->heuristic() == sg->heuristic(),
         "Heuristics do not match.");
-    std::unique_ptr<Fusion> fusion_to_run = segmented_fusion_->makeFusion(sg);
+    auto&& [ir_cloner, fusion_to_run] = segmented_fusion_->makeFusion(sg);
```

Comment:
`ir_cloner` doesn't seem to be used, so how about
```suggestion
    auto fusion_to_run = segmented_fusion_->makeFusion(sg).second;
```

Reply:
Several other places in this PR may be simplified like this.